### PR TITLE
I dont know man this feels like my little son

### PR DESCRIPTION
**Pull request by carlillous**
Haven't tested the CLI because my mental sanity is going to explode but everything should work fine. 

To be implemented: Test classes and things like that (maybe i'll do it tomorrow but anyone is free to do it).

### CODE REVIEW

File: java/Indexer/src/main/java/MapIndexer.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.List;
+import java.util.Map;
+
+public class MapIndexer {
+
+    private Map<String, HashSet<Integer>> wordIndexes;
+    private DataMart dataMart;
+
+    public MapIndexer() {
+        wordIndexes = new HashMap<>();
```
Comment (adam-prochazka) [inline java/Indexer/src/main/java/MapIndexer.java:13]: The whole `HashMap` here is unnecessary Imo. It creates another Data Mart basically. You can just use the `FileDataMart`. It automatically checks for duplicates. But perhaps I should mention it in the documentation.

**Review by adam-prochazka — CHANGES_REQUESTED** [review]
Mentioned some stuff. Take a look at it. We don't have to merge it yet. It will take longer to fix.

File: java/DataLake/src/main/java/Reader.java [hunk 2]
```diff
@@ -75,8 +77,20 @@ public Book readBook(String path) {
...
         }
+        bookName = this.extractBookName(fileName);
+        Book book = new Book(bookName,words);
+        handler.addBook(book.getIndex(), book.getName());
+
+        return book;
+    }
+
+    public static String extractBookName(String fileName) {
+        String[] parts = fileName.split("\\)");
```
[the remark this answers is not in the document]
Comment (carlillous) [reply java/DataLake/src/main/java/Reader.java:88]: files in the datalake follow this structure (GUTENBERG ID)Book name.txt. 

So what we are doing is splitting the string by the ), why wouldn't this work on other os.

File: java/Crawler/src/main/java/Controller.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    public Controller(){
+        this.start();
+    }
+
+    public void start(){
+
+        // Create a ScheduledExecutorService with a thread to execute scheduled tasks
+        ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
+
```
[the remark this answers is not in the document]
Comment (carlillous) [reply java/Crawler/src/main/java/Controller.java:17]: i asked chat gpt to execute a fuction every minute and this was the prompt i most liked. I can search for an easier way but i tried it and it worked fine.

File: java/DataLake/src/test/java/ReaderTest.java [hunk 4]
```diff
@@ -6,7 +6,7 @@ public class ReaderTest {
     @Test
     public void testReadBook() {
         Reader reader = new Reader();
-        String filePath = "src/files/pg72005.txt";
+        String filePath = "files/BOOK";
```
[the remark this answers is not in the document]
Comment (carlillous) [reply java/DataLake/src/test/java/ReaderTest.java:9]: yeah i forgor

File: java/Indexer/src/main/java/MapIndexer.java (hunk 1 above)
Comment (carlillous) [reply java/Indexer/src/main/java/MapIndexer.java:13]: you are right. ill try to change it

File: java/DataLake/src/main/java/Reader.java [hunk 5]
```diff
@@ -75,8 +77,20 @@ public Book readBook(String path) {
         } catch (IOException e) {
             e.printStackTrace();
         }
+        bookName = this.extractBookName(fileName);
+        Book book = new Book(bookName,words);
+        handler.addBook(book.getIndex(), book.getName());
+
+        return book;
```
[the remark this answers is not in the document]
Comment (carlillous) [reply java/DataLake/src/main/java/Reader.java:84]: what do you mean

[the remark this answers is not in the document]
Comment (adam-prochazka) [reply java/DataLake/src/main/java/Reader.java:84]: Ahh never mind i did not see some lines

File: java/DataLake/src/main/java/Reader.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (adam-prochazka) [reply java/DataLake/src/main/java/Reader.java:88]: Ah alright, I thought it's splitting by `\`.

**Review by JoelClemente — APPROVED** [review]
Its good for me mendozita

File: java/DataLake/src/main/java/Reader.java [hunk 6]
```diff
@@ -75,8 +78,20 @@ public Book readBook(String path) {
...
+        bookName = this.extractBookName(fileName);
+        Book book = new Book(bookName,words);
+        datalake.addBook(book.getIndex(), book.getName());
+
+        return book;
+    }
+
+    public static String extractBookName(String fileName) {
+        String[] parts = fileName.split("\\)");
+        String extractedName = parts[1].replace("_", " ");
```
Comment (adam-prochazka) [inline java/DataLake/src/main/java/Reader.java:90]: The app sometimes shuts down on this step. Don't know how to replicate it.

Comment (carlillous) [reply java/DataLake/src/main/java/Reader.java:90]: i think it should work fine if we remove the regular expression and just put ")" inside the split method @adam-prochazka

Comment (carlillous) [reply java/DataLake/src/main/java/Reader.java:90]: ok so nope there is no other way of doing it

Comment (carlillous) [reply java/DataLake/src/main/java/Reader.java:90]: We assume that the fileName follows the Crawler downloaded books structure thing like:

(GUTENBER ID) book name.txt 

it wont work if the file is not like this.

**Review by adam-prochazka — APPROVED** [review]

**Pull request merged**